daemon/cmd/endpoint.go

```diff
@@ -295,11 +295,10 @@ func (m *endpointCreationManager) DebugStatus() (output string) {
 // createEndpoint attempts to create the endpoint corresponding to the change
 // request that was specified.
 func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
+	if epTemplate.DatapathConfiguration == nil {
+		epTemplate.DatapathConfiguration = &models.EndpointDatapathConfiguration{}
+	}
 	if option.Config.EnableEndpointRoutes {
-		if epTemplate.DatapathConfiguration == nil {
-			epTemplate.DatapathConfiguration = &models.EndpointDatapathConfiguration{}
-		}
-
```

Comment:
I'm concerned that this will break chaining mode, for reasons described in commit 8da8b88c7509a1a57f928f43ffe46d676a71ca66.